protocols/raft/src/main/java/io/atomix/protocols/raft/impl/RaftContext.java

```diff
@@ -204,11 +204,17 @@ public RaftContext(
   }
 
   private void onUncaughtException(Throwable error) {
-    log.error("An uncaught exception occurred", error);
+    log.error("An uncaught exception occurred, transition to inactive role", error);
     try {
-      failureListeners.forEach(Runnable::run);
-    } catch (Exception exception) {
-      log.error("Could not notify failure listeners", exception);
+      // to prevent further operations submitted to the threadcontext to execute
+      transition(Role.INACTIVE);
+      try {
+        failureListeners.forEach(Runnable::run);
+      } catch (Exception exception) {
+        log.error("Could not notify failure listeners", exception);
+      }
+    } catch (Throwable e) {
+      onUncaughtException(e);
```

Comment:
Is this a recursive method? Could we end up in an endless loop?

Reply:
Yes. Since not able to transition or not able to notify failure listeners is a bigger problem, I thought having an endless loop here is ok. But I can eliminate recursion here.

Reply:
What happens if we have an endless loop? Just printing out errors over and over and over? Is there no way we could just let-it-crash<sup>TM</sup>? 

As a Zeebe operator/dev, when it comes to diagnostics, what's the most useful behavior here? Personally would be to let it crash, but is this executed in one of those pathways where we catch `Throwable`? :disappointed_relieved:

Reply:
If we don't catch it `ScheduledThreadPoolExecutor` catches it and proceeds with the next task. I don't know how to let it crash.